src/Altinn.Notifications.Persistence/NotificationRepository.cs

```diff
@@ -233,9 +233,9 @@ public async Task<List<Target>> GetUnsentTargets()
             return target;
         }
 
-        public async Task<Target> UpdateSentTarget(int id)
+        public async Task<Target?> UpdateSentTarget(int id)
```

Comment:
Must admit this was a bit confusing. A method named Update... that runs a select - or is the notifications.update_senttarget(@id) updating something?

Reply:
I'm not overly happy about the structure of the SQL command text either. There might be a better way, but I'm not too familiar with the quirks of PostgreSQL. 

The command creates a call to a function that updates the target and returns a row that is selected.

Reply:
Any particular reason why we use functions in the database and not in code? Or at least a call to an update statement instead of a function?

Reply:
I believe it provides better protection against SQL injection. When PostgreSQL handles the transfer of values through function parameters.

Reply:
I would assume any sql statement could be parameterized to prevent sql injection including an simple update statement. Also in this case it's an int that's passed in, so don't really see why we need the extra layer of complexity by "hiding" parts of the code in an SQL function.

Reply:
This is a fine discussion to have as a team.